src/openvic-simulation/economy/production/ProductionType.hpp

```diff
@@ -97,19 +98,19 @@ namespace OpenVic {
 		ProductionTypeManager();
 
 		bool add_production_type(
-			std::string_view identifier,
+			const std::string_view identifier,
 			std::optional<Job> owner,
-			std::vector<Job>&& employees,
-			ProductionType::template_type_t template_type,
-			Pop::pop_size_t workforce,
-			GoodDefinition::good_definition_map_t&& input_goods,
-			GoodDefinition const* output_goods,
-			fixed_point_t value,
+			std::vector<Job>&& jobs,
+			const ProductionType::template_type_t template_type,
+			const Pop::pop_size_t base_workforce_size,
+			GoodDefinition::good_definition_map_t const&& input_goods,
```

Comment:
All instances of `const&&` in this file and in ProductionType.cpp should be changed to just `&&` - this represents an r-value reference, or a "moved value". When moving something like a vector or a map, you "steal" its internal memory pointer and replace it will null so that when the original value is destroyed it won't free that memory. You can't do that if the value is const, as its internal pointer can't be changed to null, forcing you to make a copy instead and defeating the whole purpose of moving the object.

Reply:
Good advice, thanks.
I still struggle with const or no const and & vs * vs nothing.
In my mind, I think of them as functional aspects of a variable.
So in my mind const means "read only".
& I think of as "guaranteed instance".
* I treat as nullable.

This way of thinking helps me in some cases but fails to properly optimise code. I'll change it to just &&.